presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -183,6 +184,155 @@ else if (rescaleFactor > 0) {
         }
     }
 
+    public static Slice add(Slice left, Slice right)
```

Comment:
let's keep return by parameter semantics, e.g:
`add(Slice left, Slice right, Slice result)`.
This will make users of the code think about performance impact of frequently allocating result Slice object.
Also, with this code it is impossible to add in place.